pyomo/core/kernel/expr_coopr3.py

```diff
@@ -638,7 +638,11 @@ def to_string(self, ostream=None, verbose=None, precedence=0):
             ostream.write("( ")
         first = True
         if self._coef != 1:
+            if self._coef < 0:
+                ostream.write("(")
             ostream.write(str(self._coef))
+            if self._coef < 0:
+                ostream.write(")")
```

Comment:
I'd be happy for some additional eyes on it.  we may not be able to get around the "simple" fix that @gseastream put in.  However, I think a better option will be to look at the the way we determine precedence when generating the string representations.  The cases that I'd like us to look at are:
  - `sub(y, mul(-1,x))`
  - `sub(y, mul(-1,mul(x,x)))`
  - `sub(y, mul(-2,x))`
  - `sub(y, mul(-2,mul(x,x)))`
  - `sub(y, mul(2,x))`
  - `sub(y, mul(2,mul(x,x)))`
  - `sub(y, neg(x))`
  - `add(y, mul(-1,x))`
  - `add(y, mul(-1,mul(x,x)))`
  - `add(y, mul(-2,x))`
  - `add(y, mul(-2,mul(x,x)))`
  - `add(y, mul(2,x))`
  - `add(y, mul(2,mul(x,x)))`
  - `add(y, neg(x))`